### PR TITLE
[BUG] Support submits spark pipeline with reference component

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/spark_job_entry_mixin.py [hunk 1]
```diff
@@ -39,6 +39,11 @@ def entry(self, value: Union[Dict[str, str], SparkJobEntry, None]):
         self._entry = value
 
     def _validate_entry(self):
+        if self.entry is None:
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/spark_job_entry_mixin.py:42]: shall we change the msg in below if branch?

Comment (YingChen1996) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/spark_job_entry_mixin.py:42]: updated

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/spark_job_entry_mixin.py [hunk 2]
```diff
@@ -39,8 +39,13 @@ def entry(self, value: Union[Dict[str, str], SparkJobEntry, None]):
...
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/spark_job_entry_mixin.py:42]: do we have a test case covering this scenario?

Comment (YingChen1996) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/spark_job_entry_mixin.py:42]: @wangchao1230 

we need wait for doris pr: https://msdata.visualstudio.com/Vienna/_git/sdk-cli-v2/pullrequest/927066?_a=files&path=/src/mldesigner/mldesigner/_generate/_generators/_constants.py, should we add a test and skip it for now?

And it's a test with mldesigner, should we put it in sdk-cli-v2 repro or this repro

Comment (wangchao1230) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/spark_job_entry_mixin.py:42]: I see. Let's do it there then.

**Pull request merged**